spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'factories/articles'
+
+RSpec.describe ArticlesController, type: :controller do
+  let(:article) { create :article }
+
+  describe '[GET] #index' do
+    before { get :index }
+
+    it do
+      expect(assigns(:articles)).to eq([article])
+      expect(response).to render_template('index')
+    end
+  end
+
+  describe '[GET] #show' do
+    context 'when article was found' do
+      before { get :show, params: { id: article.id } }
+
+      it do
+        expect(assigns(:article)).to eq(article)
+        expect(response).to render_template('show')
+      end
+    end
+
+    context 'article was not found' do
+      before { get :show, params: { id: 10_000 } }
+
+      it do
+        expect(response).to redirect_to(articles_path)
+        expect(flash: [:danger])
```

Comment:
?

Reply:
I added this code so that when a user doesn't exists the program should redirect it to the index page and show an error message.